x-pack/plugins/kubernetes_security/public/components/container_name_widget/index.tsx

```diff
@@ -0,0 +1,227 @@
+    const { field: sortingField, direction: sortingDirection } = sort;
+
+    setSortField(sortingField);
+    setSortDirection(sortingDirection);
+  }, []);
+
+  const sorting: EuiTableSortingType<ContainerNameArrayDataValue> = {
+    sort: {
+      field: sortField as keyof ContainerNameArrayDataValue,
+      direction: sortDirection as 'desc' | 'asc',
+    },
+    enableAllColumns: true,
+  };
+
+  const { getFilterForValueButton, getFilterOutValueButton, filterManager } = useSetFilter();
+  const filterButtons = useMemo((): FilterButtons => {
+    const result: FilterButtons = {
+      filterForButtons:
+        data?.pages
+          ?.map((aggsData) => {
+            return aggsData?.buckets.map((aggData) => {
+              return getFilterForValueButton({
+                field: CONTAINER_IMAGE_NAME,
+                filterManager,
+                size: 'xs',
+                onClick: () => {},
+                onFilterAdded: () => {},
+                ownFocus: false,
+                showTooltip: true,
+                value: aggData.key as string,
+              });
+            });
+          })
+          .flat() || [],
+
+      filterOutButtons:
+        data?.pages
+          ?.map((aggsData) => {
+            return aggsData?.buckets.map((aggData) => {
+              return getFilterOutValueButton({
+                field: CONTAINER_IMAGE_NAME,
+                filterManager,
+                size: 'xs',
+                onClick: () => {},
+                onFilterAdded: () => {},
+                ownFocus: false,
+                showTooltip: true,
+                value: aggData.key as string,
+              });
+            });
+          })
+          .flat() || [],
+    };
+    return result;
+  }, [data, getFilterForValueButton, getFilterOutValueButton, filterManager]);
+
+  const containerNameArray = useMemo((): ContainerNameArrayDataValue[] => {
+    return data
+      ? data?.pages
+          ?.map((aggsData) => {
+            return aggsData?.buckets.map((aggData) => {
+              return {
+                name: aggData.key as string,
+                count: aggData.count_by_aggs.value,
+              };
+            });
+          })
+          .flat()
+      : [];
+  }, [data]);
+
+  const columns = useMemo((): Array<EuiBasicTableColumn<ContainerNameArrayDataValue>> => {
+    return [
+      {
+        field: 'name',
+        name: CONTAINER_NAME_SESSION,
+        'data-test-subj': NAME_COLUMN_TEST_ID,
+        render: (name: string) => {
+          const indexHelper = containerNameArray.findIndex((obj) => {
+            return obj.name === name;
+          });
+
+          return (
+            <ContainerNameRow
+              name={name}
+              index={indexHelper}
+              filterButtonIn={filterButtons.filterForButtons[indexHelper]}
+              filterButtonOut={filterButtons.filterOutButtons[indexHelper]}
+            />
+          );
+        },
+        align: 'left',
+        width: '216px',
```

Comment:
should use percentage value here, so that when the table wraps to a new row (when the screen is small) the table can fill the row.